### DISCUSSION TITLE
Compiled Binding

### DISCUSSION THREAD

**Discussion by danielftz** [CommunityToolkit/Maui.Markup#154] [Q&A] [answered]
So far in markup I have used .Bind() method. I know XAML has the compiled binding feature, and Im just wondering does c# markup have an equivalent of that?

**Comment by TheCodeTraveler** [comment, marked as answer]
We don't. ~But I'd love to add it if you have ideas on how to implement it for a [New Feature Proposal](https://github.com/CommunityToolkit/Maui.Markup/issues/new?assignees=&labels=new%2C+proposal&template=open-a-new-feature-proposal.yml&title=%5BProposal%5D+)!~

New Feature Proposal: https://github.com/CommunityToolkit/Maui.Markup/issues/155

**Reply by TheCodeTraveler** [reply]
I see some examples here in the .NET MAUI Unit Tests on how to consume TypedBindings in C#: https://github.com/dotnet/maui/blob/main/src/Controls/tests/Core.UnitTests/TypedBindingUnitTests.cs

**Reply by danielftz** [reply]
Thanks, I'll look into it. Keep up the good work.

**Reply by TheCodeTraveler** [reply]
Here's a quick POC on how we could implement TypedBindings in C# Markup:
- [TypedBindingExtensions.cs](https://github.com/CommunityToolkit/Maui.Markup/blob/Compiled-Bindings/src/CommunityToolkit.Maui.Markup/TypedBindingExtensions.cs)
- [Two-way TypedBinding for `RefreshView.IsRefreshingProperty`](https://github.com/CommunityToolkit/Maui.Markup/blob/d1018e92ec89e4a5c1dc1f0cfe47ceee5e9eedbc/samples/CommunityToolkit.Maui.Markup.Sample/Pages/NewsPage.cs#L30)
- [One-way (aka readonly) TypedBinding for `Label.TextProperty`](https://github.com/CommunityToolkit/Maui.Markup/blob/f6692deaa98e742086ed79d9ca695e3bd36825a9/samples/CommunityToolkit.Maui.Markup.Sample/Views/News/StoryDataTemplate.cs#L27)

**Reply by TheCodeTraveler** [reply]
Aw heck, I've gone ahead and opened a New Feature Proposal! Thanks for the inspiration!

Check it out and let me know what you think:
https://github.com/CommunityToolkit/Maui.Markup/issues/155

**Reply by danielftz** [reply]
> Aw heck, I've gone ahead and opened a New Feature Proposal! Thanks for the inspiration!
> 
> Check it out and let me know what you think: #155

tbh I dont know too much about compiled binding/ typed binding. Im gonna use your code in my project and give it a try and Ill let you know

**Reply by TheCodeTraveler** [reply]
Right on! I'll open up a Draft PR for the new API to generate an updated CommunityToolkit.Maui.Markup NuGet Package that you can try out 👍

**Comment by CliffAgius** [comment]
Closed and answered
